modules/opensles/recorder.c

```diff
@@ -107,6 +109,49 @@ static int createAudioRecorder(struct ausrc_st *st, struct ausrc_prm *prm)
 		return ENODEV;
 	}
 
+	SLAndroidConfigurationItf player_config;
+	r = (*st->recObject)->GetInterface(st->recObject,
+											SL_IID_ANDROIDCONFIGURATION,
+											&player_config);
+	if (SL_RESULT_SUCCESS == r) {
+		SLint32 stream_type = SL_ANDROID_RECORDING_PRESET_VOICE_COMMUNICATION;
+
+		(*player_config)->SetConfiguration(player_config,
+										   SL_ANDROID_KEY_RECORDING_PRESET, &stream_type, sizeof(SLint32));
+
+        SLuint32 performance = SL_ANDROID_PERFORMANCE_NONE;
+        (*player_config)->SetConfiguration(player_config,
+										   SL_ANDROID_KEY_PERFORMANCE_MODE, &performance, sizeof(SLuint32));
+	}
+
+
+    SLAndroidAcousticEchoCancellationItf player_aec;
+    r = (*st->recObject)->GetInterface(st->recObject,
+                                            SL_IID_ANDROIDACOUSTICECHOCANCELLATION,
+                                            &player_aec);
+    if (SL_RESULT_SUCCESS == r) {
+        SLboolean enable = SL_BOOLEAN_TRUE;
+        (*player_aec)->SetEnabled(player_aec, enable);
+    }
```

Comment:
should AEC always be enabled ?

There are cases where the AEC is provided by software,
or the user wants to disable AEC on purpose.